src/main/resources/db/migration/all/20241104135401__add_offline_app_to_cas1_space_booking.sql

```diff
@@ -0,0 +1,8 @@
+ALTER TABLE cas1_space_bookings ALTER COLUMN approved_premises_application_id DROP NOT NULL;
+ALTER TABLE cas1_space_bookings ADD offline_application_id UUID NULL;
+
+ALTER TABLE cas1_space_bookings ADD FOREIGN KEY (offline_application_id) REFERENCES offline_applications(id);
+
+alter table cas1_space_bookings
+    add constraint only_one_application_set
+    check ((approved_premises_application_id is null) <> (offline_application_id is null));
```

Comment:
Does this fall over if both are set?

Reply:
Yep that's the idea. We make that assumption in code (one is set) so it's nice to have a database level constraint to guarantee that (i've never used checks before, they're handy!)

Reply:
I was thinking this DB check doesn't work if a space booking is linked to an application id and an offline application id, this would mean:

check (FALSE <> FALSE)

Which returns FALSE so not really what we want?  Do we not want an XOR type check?

Reply:
The intention _is_ an XOR check, which i believe is what we're getting here e.g. a space booking must be linked to  an application or an offline application (one and only one)